### PR TITLE
Update to Viash 0.9

### PR DESCRIPTION
**Pull request by rcannood**
## Checklist before requesting a review
- [x] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality (new method, new metric, ...)
  - [x] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [x] Proposed changes are described in the CHANGELOG.md

- [x] CI Tests succeed and look good!

### CODE REVIEW

File: src/control_methods/ground_truth/config.vsh.yaml [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
+  summary: "Returns the ground truth predictions."
+  description: |
+    The identity function that returns the ground-truth information as the output.
+  preferred_normalization: counts
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/control_methods/ground_truth/config.vsh.yaml:14]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

File: src/control_methods/mean_across_celltypes/config.vsh.yaml [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
+  summary: Baseline method that returns mean of cell type's outcomes
+  description: |
+    Baseline method that predicts for a cell type the mean of its outcomes of all compounds.
+resources:
+  - type: python_script
+    path: script.py
+  - path: ../../utils/anndata_to_dataframe.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/control_methods/mean_across_celltypes/config.vsh.yaml:14]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/control_methods/mean_across_compounds/config.vsh.yaml [hunk 3]
```diff
@@ -1,21 +1,22 @@
...
+  summary: Baseline method that returns mean of compound's outcomes
+  description: |
+    Baseline method that predicts for a compound the mean of its outcomes of all samples.
+resources:
+  - type: python_script
+    path: script.py
+  - path: ../../utils/anndata_to_dataframe.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/control_methods/mean_across_compounds/config.vsh.yaml:14]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/control_methods/mean_outcome/config.vsh.yaml [hunk 4]
```diff
@@ -1,21 +1,22 @@
...
+  summary: Baseline method that returns mean of gene's outcomes
+  description: |
+    Baseline method that predicts for a gene the mean of its outcomes of all samples.
+resources:
+  - type: python_script
+    path: script.py
+  - path: ../../utils/anndata_to_dataframe.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/control_methods/mean_outcome/config.vsh.yaml:14]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/control_methods/sample/config.vsh.yaml [hunk 5]
```diff
@@ -1,23 +1,24 @@
...
+  summary: Sample predictions from the training data
+  description: |
+    This method samples the training data to generate predictions.
+  preferred_normalization: counts
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/control_methods/sample/config.vsh.yaml:14]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

File: src/control_methods/zeros/config.vsh.yaml [hunk 6]
```diff
@@ -1,20 +1,21 @@
...
+  label: Zeros
+  summary: Baseline method that predicts all zeros
+  description: |
+    Baseline method that predicts all zeros.
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/control_methods/zeros/config.vsh.yaml:13]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/methods/jn_ap_op2/config.vsh.yaml [hunk 7]
```diff
@@ -1,36 +1,35 @@
...
+    default: [dl40, dl200]
+    info:
+      test_default: [dl40]
+resources:
+  - type: python_script
+    path: script.py
+  - path: helper.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/jn_ap_op2/config.vsh.yaml:34]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/methods/lgc_ensemble_direct/config.vsh.yaml [hunk 8]
```diff
@@ -1,62 +1,60 @@
...
+  - path: ../lgc_ensemble_helpers/predict.py
+  - path: ../lgc_ensemble_helpers/prepare_data.py
+  - path: ../lgc_ensemble_helpers/train.py
+  - path: ../lgc_ensemble_helpers/divisor_finder.py
+  - path: ../../utils/anndata_to_dataframe.py
     
-platforms:
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/lgc_ensemble_direct/config.vsh.yaml:57]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/methods/lgc_ensemble_predict/config.vsh.yaml [hunk 9]
```diff
@@ -1,41 +1,39 @@
...
+  - path: ../lgc_ensemble_helpers/predict.py
+  - path: ../lgc_ensemble_helpers/prepare_data.py
+  - path: ../lgc_ensemble_helpers/train.py
+  - path: ../lgc_ensemble_helpers/divisor_finder.py
+  - path: ../../utils/anndata_to_dataframe.py
     
-platforms:
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/lgc_ensemble_predict/config.vsh.yaml:36]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/methods/lgc_ensemble_prepare/config.vsh.yaml [hunk 10]
```diff
@@ -1,64 +1,62 @@
...
+  - path: ../lgc_ensemble_helpers/predict.py
+  - path: ../lgc_ensemble_helpers/prepare_data.py
+  - path: ../lgc_ensemble_helpers/train.py
+  - path: ../lgc_ensemble_helpers/divisor_finder.py
+  - path: ../../utils/anndata_to_dataframe.py
     
-platforms:
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/lgc_ensemble_prepare/config.vsh.yaml:59]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/methods/lgc_ensemble_train/config.vsh.yaml [hunk 11]
```diff
@@ -1,54 +1,52 @@
...
+  - path: ../lgc_ensemble_helpers/predict.py
+  - path: ../lgc_ensemble_helpers/prepare_data.py
+  - path: ../lgc_ensemble_helpers/train.py
+  - path: ../lgc_ensemble_helpers/divisor_finder.py
+  - path: ../../utils/anndata_to_dataframe.py
     
-platforms:
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/lgc_ensemble_train/config.vsh.yaml:49]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/methods/nn_retraining_with_pseudolabels/config.vsh.yaml [hunk 12]
```diff
@@ -1,45 +1,44 @@
...
+  - type: python_script
+    path: script.py
+  - path: notebook_264.py
+  - path: notebook_266.py
+  - path: ../../utils/anndata_to_dataframe.py
 
-platforms:
+engines:
   - type: docker
     image: nvcr.io/nvidia/tensorflow:24.03-tf2-py3
```
Comment (KaiWaldrant) [inline src/methods/nn_retraining_with_pseudolabels/config.vsh.yaml:43]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/tensorflow_nvidia:1.1.0
```

File: src/methods/pyboost/config.vsh.yaml [hunk 13]
```diff
@@ -1,43 +1,42 @@
...
+    info:
+      test_default: [knn_recommender]
+resources:
+  - type: python_script
+    path: script.py
+  - path: helper.py
+  - path: ../../utils/anndata_to_dataframe.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/pyboost/config.vsh.yaml:41]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/methods/scape/config.vsh.yaml [hunk 14]
```diff
@@ -1,68 +1,67 @@
...
+    description: The minimum number of top drugs to consider.
+    default: 50
+    info:
+      test_default: 0
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: nvcr.io/nvidia/tensorflow:24.03-tf2-py3
```
Comment (KaiWaldrant) [inline src/methods/scape/config.vsh.yaml:66]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/tensorflow_nvidia:1.1.0
```

File: src/methods/transformer_ensemble/config.vsh.yaml [hunk 15]
```diff
@@ -1,50 +1,49 @@
...
+    description: "Number of epochs to wait for early stopping."
+resources:
+  - type: python_script
+    path: script.py
+  - path: models.py
+  - path: utils.py
+  - path: train.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_pytorch_nvidia:1.0.4
```
Comment (KaiWaldrant) [inline src/methods/transformer_ensemble/config.vsh.yaml:48]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/pytorch_nvidia:1.1.0
```

File: src/metrics/mean_rowwise_correlation/config.vsh.yaml [hunk 16]
```diff
@@ -1,65 +1,66 @@
...
+      documentation_url: null
+      min: -1
+      max: 1
+      maximize: true
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/metrics/mean_rowwise_correlation/config.vsh.yaml:58]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

File: src/metrics/mean_rowwise_error/config.vsh.yaml [hunk 17]
```diff
@@ -1,49 +1,50 @@
...
+      documentation_url: null
+      min: 0
+      max: "+inf"
+      maximize: false
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/metrics/mean_rowwise_error/config.vsh.yaml:42]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

File: src/process_dataset/add_uns_metadata/config.vsh.yaml [hunk 18]
```diff
@@ -1,58 +1,59 @@
...
+    type: file
+    required: true
+    direction: output
+    example: resources/datasets/neurips-2023-data/pseudobulk_uns.h5ad
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/add_uns_metadata/config.vsh.yaml:54]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/process_dataset/bootstrap/config.vsh.yaml [hunk 19]
```diff
@@ -1,61 +1,62 @@
...
+      - name: --var_replace
+        type: boolean
+        default: true
+        description: Whether to sample with replacement.
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/bootstrap/config.vsh.yaml:57]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/process_dataset/compute_pseudobulk/config.vsh.yaml [hunk 20]
```diff
@@ -1,33 +1,34 @@
...
+    type: file
+    required: true
+    direction: output
+    example: resources/datasets/neurips-2023-data/pseudobulk.h5ad
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/compute_pseudobulk/config.vsh.yaml:26]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/process_dataset/convert_h5ad_to_parquet/config.vsh.yaml [hunk 21]
```diff
@@ -1,49 +1,50 @@
...
+    required: true
+    direction: output
+    example: resources/datasets/neurips-2023-data/id_map.csv
+resources:
+  - type: python_script
+    path: script.py
+  - path: ../../utils/anndata_to_dataframe.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/convert_h5ad_to_parquet/config.vsh.yaml:42]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/process_dataset/convert_kaggle_h5ad_to_parquet/config.vsh.yaml [hunk 22]
```diff
@@ -1,87 +1,88 @@
...
-      path: script.py
-platforms:
+        direction: output
+        example: resources/neurips-2023-kaggle/id_map.csv
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/convert_kaggle_h5ad_to_parquet/config.vsh.yaml:80]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/process_dataset/convert_kaggle_h5ad_to_parquet/config.vsh.yaml [hunk 23]
```diff
@@ -1,87 +1,88 @@
...
     setup:
       - type: python
         packages: [ fastparquet, anndata, pandas ]
```
Comment (KaiWaldrant) [inline src/process_dataset/convert_kaggle_h5ad_to_parquet/config.vsh.yaml:83]:
```suggestion
        packages: [ fastparquet, pandas ]
```

File: src/process_dataset/filter_obs/config.vsh.yaml [hunk 24]
```diff
@@ -1,33 +1,34 @@
...
+    type: file
+    required: true
+    direction: output
+    example: resources/datasets/neurips-2023-data/sc_counts_cleaned.h5ad
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/filter_obs/config.vsh.yaml:26]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

File: src/process_dataset/filter_vars/config.vsh.yaml [hunk 25]
```diff
@@ -1,33 +1,34 @@
...
+    type: file
+    required: true
+    direction: output
+    example: resources/datasets/neurips-2023-data/pseudobulk_cleaned.h5ad
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/filter_vars/config.vsh.yaml:26]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

File: src/process_dataset/generate_id_map/config.vsh.yaml [hunk 26]
```diff
@@ -1,33 +1,34 @@
...
+    type: file
+    required: true
+    direction: output
+    example: resources/datasets/neurips-2023-data/id_map.csv
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_python:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/generate_id_map/config.vsh.yaml:26]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
```

File: src/process_dataset/generate_id_map/config.vsh.yaml [hunk 27]
```diff
@@ -1,33 +1,34 @@
...
     setup:
       - type: python
         packages: [ anndata ]
```
Comment (KaiWaldrant) [inline src/process_dataset/generate_id_map/config.vsh.yaml:29]:
```suggestion
```

File: src/process_dataset/run_limma/config.vsh.yaml [hunk 28]
```diff
@@ -1,55 +1,56 @@
...
+  - name: --control_compound
+    type: string
+    required: false
+    default: "Dimethyl Sulfoxide"
+resources:
+  - type: r_script
+    path: script.R
+engines:
   - type: docker
     image: ghcr.io/openproblems-bio/base_r:1.0.4
```
Comment (KaiWaldrant) [inline src/process_dataset/run_limma/config.vsh.yaml:48]:
```suggestion
    image: ghcr.io/openproblems-bio/base_images/r:1.1.0
```

**Review by KaiWaldrant — APPROVED** [review]
minor suggestions

File: src/process_dataset/convert_h5ad_to_parquet/config.vsh.yaml [hunk 29]
```diff
@@ -39,7 +39,7 @@ resources:
   - path: ../../utils/anndata_to_dataframe.py
 engines:
   - type: docker
-    image: ghcr.io/openproblems-bio/base_python:1.0.4
+    image: ghcr.io/openproblems-bio/base_images/python:1.1.0
     setup:
       - type: python
         packages: [ fastparquet, anndata, pandas ]
```
Comment (KaiWaldrant) [inline src/process_dataset/convert_h5ad_to_parquet/config.vsh.yaml:45]:
```suggestion
      packages: [ fastparquet, pandas ]
```

File: src/methods/scape/config.vsh.yaml [hunk 30]
```diff
@@ -63,7 +63,7 @@ resources:
     path: script.py
 engines:
   - type: docker
-    image: nvcr.io/nvidia/tensorflow:24.03-tf2-py3
+    image: ghcr.io/openproblems-bio/base_images/tensorflow_nvidia:1.1.0
     setup:
       - type: apt
```
Comment (KaiWaldrant) [inline src/methods/scape/config.vsh.yaml:68]:
```suggestion
```

File: src/methods/scape/config.vsh.yaml [hunk 31]
```diff
@@ -63,7 +63,7 @@ resources:
...
         packages:
```
Comment (KaiWaldrant) [inline src/methods/scape/config.vsh.yaml:69]:
```suggestion
```

File: src/methods/nn_retraining_with_pseudolabels/config.vsh.yaml [hunk 32]
```diff
@@ -1,47 +1,46 @@
...
+  - path: ../../utils/anndata_to_dataframe.py
 
-platforms:
+engines:
   - type: docker
-    image: nvcr.io/nvidia/tensorflow:24.03-tf2-py3
+    image: ghcr.io/openproblems-bio/base_images/tensorflow_nvidia:1.1.0
     setup:
       - type: apt
         packages:
```
Comment (KaiWaldrant) [inline src/methods/nn_retraining_with_pseudolabels/config.vsh.yaml:46]: remove packages from line 45 - 54

File: src/methods/scape/config.vsh.yaml [hunk 33]
```diff
@@ -1,70 +1,69 @@
...
+      test_default: 0
+resources:
+  - type: python_script
+    path: script.py
+engines:
   - type: docker
-    image: nvcr.io/nvidia/tensorflow:24.03-tf2-py3
+    image: ghcr.io/openproblems-bio/base_images/tensorflow_nvidia:1.1.0
     setup:
       - type: apt
```
Comment (KaiWaldrant) [inline src/methods/scape/config.vsh.yaml:68]: remove pacakges from line 68 - 77

**Review by KaiWaldrant — APPROVED** [review]

File: src/methods/scape/config.vsh.yaml (hunk 33 above)
Comment (rcannood) [reply src/methods/scape/config.vsh.yaml:68]: thanks!

File: src/methods/nn_retraining_with_pseudolabels/config.vsh.yaml (hunk 32 above)
Comment (rcannood) [reply src/methods/nn_retraining_with_pseudolabels/config.vsh.yaml:46]: thanks!

**Pull request merged**